adhocracy4/comments_async/static/comments_async/comment_form.jsx

```diff
@@ -112,12 +124,24 @@ export default class CommentForm extends React.Component {
               style={textareaStyle}
             />
             <div className="row">
-              <label htmlFor="id-comment-form" className="col-6 a4-comments__char-count">{this.state.commentCharCount}/4000{translated.characters}</label>
-              <div className="a4-comments__submit d-flex col-6">
-                {this.props.commentCategoryChoices
-                  ? <button type="submit" value={translated.post} className="btn a4-comments__submit-input ms-auto">{translated.post}</button>
-                  : <button type="submit" value={translated.comment} className="btn a4-comments__submit-input ms-auto">{translated.post}</button>}
-              </div>
+              <label
+                htmlFor="id-comment-form"
+                className="col-6 a4-comments__char-count"
+              >
+                {this.state.commentCharCount}/4000{translated.characters}
+              </label>
+              {!this.props.useTermsOfUse || this.props.agreedTermsOfUse
+                ? (
+                  <div className="a4-comments__submit d-flex col-6">
+                    {actionButton}
+                  </div>)
+                : (
+                  <TermsOfUseCheckbox
+                    id={`terms-of-use-checkbox-${typeof this.props.parentIndex === 'number' ? this.props.parentIndex : 'rootForm'}`}
+                    onChange={val => this.setState({ agreedTermsOfUse: val })}
+                    actionButton={actionButton}
+                  />
+                  )}
```

Comment:
Would it also work to remove the button from the TermsOfUseCheckbox component and always show the same actionButton here that is disabled if useTermsOfUse and not agreedTermsOfUse? And then only have the TermsOfUseCheckbox after the condition `!this.props.useTermsOfUse || this.props.agreedTermsOfUse` ?

Reply:
the problem here was rather the position of the button. in case of terms of use being used, it is not on the same level as the counter `.../4000`. does that make sense @Rineee ?

Reply:
@khamui @Rineee I think simplifying the js here should take priority, to solve the level issue I think we can take `col-6` class from char count label and make container div with that class for the char count and checkbox, and leaving the action button out of it. This is better practice too then having a structure class on a col. Dies that make sense?